contracts/lbp/LBPWrapper.sol

```diff
@@ -41,54 +44,34 @@ contract LBPWrapper {
         owner = _newOwner;
     }
 
-    /**
-     * @dev                       initialize lbp wrapper contract
-     * @param _LBPFactory         LBP factory address
-     * @param _swapFeePercentage  swap fee percentage
-     */
-    function initialize(address _LBPFactory, uint256 _swapFeePercentage)
-        public
-    {
-        owner = msg.sender;
-        LBPFactory = ILBPFactory(_LBPFactory);
-        swapFeePercentage = _swapFeePercentage;
-        isInitialized = true;
-    }
-
     /**
      * @dev                        initialize lbp wrapper contract
+     * @param _LBPFactory          LBP factory address
      * @param _name                LBP name
      * @param _symbol              LBP symbol
      * @param _tokens              array of tokens sorted for the LBP
-     * @param _amounts             array of initial amounts used to join pool
      * @param _weights             array of start weights for respective tokens
      * @param _swapEnabledOnStart  enable or disable swap
      * @param _startTime           start time
      * @param _endTime             end time
      * @param _endWeights          array of end weights for respective tokens
-     * @param _admin               address of admin/owner of LBP
-     * @param _userData            extra data required by LBP
      */
-    function deployLbpFromFactory(
+    function initializeLBP(
+        address _LBPFactory,
         string memory _name,
         string memory _symbol,
         IERC20[] memory _tokens,
-        uint256[] memory _amounts,
         uint256[] memory _weights,
         bool _swapEnabledOnStart,
```

Comment:
For my understanding: So `initializeLBP` creates an LBP, but the LBP does not yet contain any tokens, correct? If this is correct, I'm wondering if we should set `_swapEnabledOnStart` to `false` per default, because my understanding is that if the pool is empty no user can in fact swap. Am I missing something?

Reply:
We might then set `swapEnabledOnStart` to true in the `joinPool` function, I guess?

Reply:
Yes so we talked about this as well with @codebuster22  but this was a question we still meant to ask during the workshop. I've written it down but indeed, this is what we thought as well

Reply:
I think I will just change it like you suggested and change it back if needed after a workshop. This way makes more sense

Reply:
One of the other issues was to implement a `setPause()` function. I implemented that ticket also because we needed a way to pause/unpause anyway